Library/Homebrew/diagnostic.rb

```diff
@@ -416,7 +416,7 @@ def check_xcode_up_to_date
       end
 
       def check_for_osx_gcc_installer
-        return unless MacOS.version < "10.7" || MacOS::Xcode.version > "4.1"
+        return unless MacOS.version < "10.7" || (!MacOS::Xcode.version.nil? && MacOS::Xcode.version > "4.1")
```

Comment:
Your version is more explicit, but I think in this case readability will benefit by using either:

- `(MacOS::Xcode.version && MacOS::Xcode.version > "4.1")` or
- `(MacOS::Xcode.version || "0") > "4.1"`.